packages/edit-site/src/components/page-patterns/style.scss

```diff
@@ -104,6 +104,21 @@
 	}
 }
 
+.dataviews-view-grid {
+	.edit-site-patterns__pattern-title {
+		display: -webkit-box;
+		-webkit-line-clamp: 2;
+		-webkit-box-orient: vertical;
+		white-space: normal;
+		padding-top: 3px;
+		align-self: start;
+	}
+
+	.edit-site-patterns__pattern-lock-icon {
+		align-self: start;
+	}
+}
+
```

Comment:
This may be a personal preference, but you may find the following writing style easier to read:

```scss
.edit-site-patterns__pattern-title {
  .dataviews-view-grid & {
  }
}

.edit-site-patterns__pattern-lock-icon {
  .dataviews-view-grid & {
  }
}
```

It's similar to the writing style here:

https://github.com/WordPress/gutenberg/blob/b202059f03fafb82d48129db8aa155006b2d791f/packages/edit-site/src/components/page-templates/style.scss#L26-L38